qtfred/ui/ShipWeaponsDialog.ui

```diff
@@ -0,0 +1,223 @@
+       </property>
+      </widget>
+     </item>
+     <item>
+      <spacer name="spacerMiddle">
+       <property name="orientation">
+        <enum>Qt::Horizontal</enum>
+       </property>
+       <property name="sizeHint" stdset="0">
+        <size>
+         <width>40</width>
+         <height>20</height>
+        </size>
+       </property>
+      </spacer>
+     </item>
+     <item>
+      <widget class="QGroupBox" name="groupBoxBanks">
+       <property name="title">
+        <string>Banks</string>
+       </property>
+       <layout class="QHBoxLayout" name="BanksLayout">
+        <item>
+         <widget class="fso::fred::bankTree" name="treeBanks">
+          <property name="sizeAdjustPolicy">
+           <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
+          </property>
+          <property name="dragDropMode">
+           <enum>QAbstractItemView::DropOnly</enum>
+          </property>
+          <property name="selectionMode">
+           <enum>QAbstractItemView::MultiSelection</enum>
+          </property>
+          <property name="selectionBehavior">
+           <enum>QAbstractItemView::SelectItems</enum>
+          </property>
+          <attribute name="headerStretchLastSection">
+           <bool>false</bool>
+          </attribute>
+         </widget>
+        </item>
+       </layout>
+      </widget>
+     </item>
+    </layout>
+   </item>
+   <item>
+    <layout class="QHBoxLayout" name="AILayout">
+     <item>
+      <widget class="QComboBox" name="AICombo"/>
+     </item>
+     <item>
+      <widget class="QPushButton" name="AIButton">
+       <property name="text">
+        <string>Change AI</string>
+       </property>
+      </widget>
+     </item>
+    </layout>
+   </item>
+   <item>
+    <layout class="QHBoxLayout" name="layoutBottom">
+     <item>
+      <spacer name="spacerBottom">
+       <property name="orientation">
+        <enum>Qt::Horizontal</enum>
+       </property>
+       <property name="sizeHint" stdset="0">
+        <size>
+         <width>40</width>
+         <height>20</height>
+        </size>
+       </property>
+      </spacer>
+     </item>
+     <item>
+      <widget class="QPushButton" name="buttonClose">
+       <property name="text">
+        <string>Close</string>
+       </property>
+      </widget>
+     </item>
+    </layout>
+   </item>
+  </layout>
+ </widget>
+ <customwidgets>
+  <customwidget>
+   <class>fso::fred::weaponList</class>
+   <extends>QListView</extends>
+   <header>ui/widgets/weaponList.h</header>
+  </customwidget>
+  <customwidget>
+   <class>fso::fred::bankTree</class>
+   <extends>QTreeView</extends>
+   <header>ui/widgets/bankTree.h</header>
+  </customwidget>
+ </customwidgets>
+ <resources>
+  <include location="../resources/resources.qrc"/>
+ </resources>
+ <connections>
+  <connection>
+   <sender>buttonClose</sender>
+   <signal>clicked()</signal>
+   <receiver>fso::fred::dialogs::ShipWeaponsDialog</receiver>
+   <slot>accept()</slot>
```

Comment:
These connections in Qt are super easy to miss when debugging. What do you think of adopting the model I used in the asteroid editor rewrite instead? Remove this connection and add to the dialog header an `on_yourOkAndCancelButtonsName_accepted()` (or _rejected()) and those call the relevant accept/reject/apply/whatever methods as needed?

Again, I'll leave the decision to do this now or later up to you.